src/components/vm/disks/diskAdd.jsx

```diff
@@ -509,11 +508,39 @@ export class AddDiskModalBody extends React.Component {
             break;
         }
         case 'file': {
+            this.setState({ file: value });
+
             if (value.endsWith(".iso")) {
                 // use onValueChange instead of setState in order to perform subsequent state change logic
                 this.onValueChanged("device", "cdrom");
             }
-            this.setState({ file: value });
+
+            if (value) {
+                // Limit process time to 2 seconds and ram to 100 MB
+                cockpit.spawn(["/bin/bash", "-ec", "ulimit -t 2 -v 100000; xargs -0n1 qemu-img info --output json "], { err: "message" })
+                        .input(value)
+                        .then(output => {
+                            try {
+                                const data = JSON.parse(output);
+                                if (data['backing-filename']) {
+                                    this.dialogErrorSet(_("Importing an image with a backing file is unsupported"),
+                                                        cockpit.format(_("Backing file: $0"), data['backing-filename']));
+                                    this.setState({ customDiskVerified: false });
+                                } else {
+                                    this.setState({ customDiskVerified: true, format: data.format, dialogError: "" });
+                                }
+                            } catch (err) {
+                                this.setState({ customDiskVerified: false });
+                                console.error("qemu-img provided invalid JSON", err);
+                            }
```

Comment:
Sorry, but I realized there might be one more issue. It seems `qemu-img info` requires an exclusive lock when reading a disk. This however is not consistent with the other feature we support: disk sharing between multiple VMs.
How to reproduce:

1. Have a "raw" disk attached to running VM1, and set its access to "Writeable and shared":
2. Try to attach the same disk to VM2.
3. `qemu-img info` will fail because it could not get lock:
```
could not execute qemu-img qemu-img: Could not open '/var/lib/libvirt/images/disk1.img': Failed to get shared "write" lock
Is another process using the image [/var/lib/libvirt/images/disk1.img]? 3 [diskAdd.jsx:535:36](webpack://machines/src/components/vm/disks/diskAdd.jsx)
```

Reply:
I'm thinking how to work around this and I think of 2 solutions:

1. When `qemu-img info` fails, we could just fallback to the old way and use "raw" format. This would be simple solution (disk sharing only works with "raw" format, so it would work for above mentioned case)
2. When `qemu-img info` fails, show format selector and allow user to specify format manually (this would be a nice fallback for other unforeseeable cases, not just disk sharing, but is more complex).

Reply:
Ugh, this is serious. I didn't expect `qemu-img info` to open the file in write mode, or to even lock the file. It still doesn't make sense to add a qcow file as raw or vice versa, so IMHO falling back to letting the user decide is unacceptable: The user has no freedom in this decision, the format of the file is what it is.

If there really is no sane way to use `qemu-img info`, then let's just read it ourselves after all. Let's start with reading the first 16 bytes (with `dd`) into a binary channel, check the qcow magic, and just reject an image with a backing file. With that, we don't have to parse the backing file name in the first version. This will at least fix self-contained qcow files, which are certainly a good portion of the use case here. We can then refine this later, by filing a qemu-img info bug report ("stop the crap and just do what we ask"), doing more parsing ourselves, or perhaps resorting to `file` after all.

Reply:
@skobyda : How does it make sense for two *running* VMs to access the same qcow image? This is begging for data corruption. I thought this was only supported as long as at most one VM was running at a time? Then there shouldn't be any write lock contention. From that aspect, the current behaviour of failing image detection is even desirable. I still reject the idea of opening the file for writing just for format detection, though -- that is crazy.

Reply:
> @skobyda : How does it make sense for two _running_ VMs to access the same qcow image?

This issue is not  about qcow, as you cannot share qcow2 image. But it's a valid use case with `raw` images, which even have <shareable /> flag for that: https://libvirt.org/formatdomain.html#hard-drives-floppy-disks-cdroms , which tells that VMs will not try to gain lock when writing so disk, allowing it to be written to concurrently by multiple VMs. 
1. One use case is for example building clusters. You would have to have cluster filesystem set up for that). (more about that [here](https://www.humblec.com/how-to-share-a-disk-to-more-than-one-vm-using-libvirt-or-what-sharable-flag-do-in-libvirt/) . c-machines supports that as you can set "shareable" flag on raw disks when you click on "Edit' button next to an already attached disk.
2. The other use case is that you have one VM writing data to a raw disk, and then you want to attach the raw disk to a other VM only as a read-only so it can read from it. (you can do that by attaching disk and then editing it's configuration to "readonly")

In both cases however, with this PR, when attaching the disk to the second VM (even if the second VM is shut off, and only first VM is running), qemu-img info will fail to gain lock, and thus user will be unable to attach the disk to the second VM altogether.

Reply:
Interesting, then that would give us a path for the next step: Do a first coarse detection ourselves for qcow magic and existence of an overlay; and if that is present, call qemu-img info (as then we can't have shared images anyway). But still, I'd do that in a follow-up, given how long this has been blocked already.

Reply:
Created an upstream issue for the write lock issue https://gitlab.com/qemu-project/qemu/-/issues/1287